docs/extension.md

```diff
@@ -0,0 +1,88 @@
+# BDK Extension Model
+> :bulb: since `2.6`
+
+## Overview
+The BDK extension model consists of a single, coherent concept: the `BdkExtension` API. Note, however, that `BdkExtension` 
+itself is just a marker interface.
+
+The `BdkExtension` API is available through the module `:symphony-bdk-extension-api` but other modules might be required
+depending on what your extension needs to use.
+
+## Registering Extensions
+Extensions are registered _programmatically_ via the `ExtensionService`:
+```java
+class ExtensionExample {
+    
+    public static void main(String[] args) {
+        // using the ExtensionService
+        final SymphonyBdk bdk = new SymphonyBdk(loadFromSymphonyDir("config.yaml"));
+        bdk.extensions().register(MyBdkExtension.class);
+        
+        // or using the SymphonyBdkBuilder
+        final SymphonyBdk bdk = SymphonyBdk.builder()
+                .config(loadFromSymphonyDir("config.yaml"))
+                .extension(MyBdkExtension.class) 
+                .build();
+    }
+}
+```
+
+### Registering Extensions in Spring Boot
+To use your extension in the [BDK Spring Boot Starter](./spring-boot/core-starter.md), you simply need to register your 
```

Comment:
Should `MyBdkExtension` implement interface `BdkExtension` so that it is registered? I would assume so. If so, it would be nice to specify it here.

Reply:
Done ✅